export_dagmc_cmd/iGeom_funcs.cpp

```diff
@@ -0,0 +1,1664 @@
+      rfi->num_ref_volumes() == 0 &&
+      rfi->num_ref_faces() == 0 &&
+      rfi->num_ref_edges() == 0 &&
+      rfi->num_ref_vertices() == 0)
+    rfi->reset_ids();
+
+  RETURN(iBase_SUCCESS);
+}
+
+// This uses CGM, so might need a new way to check errors.
+void
+iGeom_getDescription( char* description_buffer,
+                      int description_buffer_length )
+{
+    CGM_iGeom_getLastErrorDesc(description_buffer, description_buffer_length);
+
+    std::string lastErrorDesc = "No Error";
+    if (description_buffer && description_buffer_length > 0) {
+        lastErrorDesc.copy( description_buffer, description_buffer_length );
+        if (lastErrorDesc.length() < (unsigned)description_buffer_length)
+            description_buffer[lastErrorDesc.length()] = '\0';
+    }
+}
+
+void
+iGeom_getEntBoundBox( /*in*/ iBase_EntityHandle entity_handle,
+                      /*out*/ double* min_x,
+                      /*out*/ double* min_y,
+                      /*out*/ double* min_z,
+                      /*out*/ double* max_x,
+                      /*out*/ double* max_y,
+                      /*out*/ double* max_z,
+                      int* err )
+{
+  RefEntity* entity = (RefEntity*)entity_handle;
+  CubitVector minc, maxc;
+  CubitStatus status = iGeom_bounding_box( entity, minc, maxc );
+  minc.get_xyz( *min_x, *min_y, *min_z );
+  maxc.get_xyz( *max_x, *max_y, *max_z );
+  RETURN( (status == CUBIT_SUCCESS ? iBase_SUCCESS : iBase_FAILURE) );
+}
+
+void
+iGeom_getEntities( /*in*/ iBase_EntitySetHandle set_handle,
+                   /*in*/ int gentity_type,
+                   /*out*/ iBase_EntityHandle **gentity_handles,
+                   int *gentity_handles_allocated,
+                   int *gentity_handles_size,
+                   int* err )
+{
+  if (RefGroup *this_set = reinterpret_cast<RefGroup*>(set_handle)) {
+    static DLIList<CubitEntity*> centities;
+    centities.clean_out();
+    this_set->get_child_entities(centities);
+    copy_ibase_type( gentity_type, centities, 
+                     gentity_handles,
+                     gentity_handles_allocated,
+                     gentity_handles_size,
+                     err );
+  }
+  else {
+    static DLIList<RefEntity*> dim_entities;
+    dim_entities.clean_out();
+    append_all_ibase_type( gentity_type, dim_entities, err );
+    if (iBase_SUCCESS != *err)
+      return;
+    
+    iGeomArrayManager gentity_handles_manager ( reinterpret_cast<void**>(gentity_handles), *(gentity_handles_allocated), *(gentity_handles_size), dim_entities.size(), sizeof(**gentity_handles) );
+    gentity_handles_manager.keep_array();
+    dim_entities.copy_to((RefEntity**)*gentity_handles);
+    RETURN(iBase_SUCCESS);
+  }  
+}
+
+void
+iGeom_getNumOfType( /*in*/ iBase_EntitySetHandle set_handle,
+                    /*in*/ int gentity_type,
+                    int* count,
+                    int* err )
+{
+  const RefGroup *this_set = reinterpret_cast<RefGroup*>(set_handle);
```

Comment:
I personnel prefer to put the '*' close to the type:
`const RefGroup* this_set`